adventurine.rb

```diff
@@ -1,26 +1,87 @@
 require 'pp' #so the board prints prettier
 
 board = [
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"]
 ]
 
-user_pos = [0,0]
-a = [-1,0]
-s = [0,1]
-w = [0,-1]
-d = [1,0]
+user_x = 0 
+user_y = 0
+valid_moves = ['w', 'a', 's', 'd', 'x', 'p']
 
-pp board
-print "You are in the northwestern most corner of a dungeon. Use 'a' to move west, 's' to move south, 'w' to move north, and 'd' to move east: "
-user_input = gets.chomp.downcase
+puts "What is your character's name?"
+character = gets.chomp
 
-if user_input == 'a' 
-  user_pos = user_pos.zip(a).map { |x, y| x + y } #if input is a then add user_pos array to a array and store it in user_pos
-elsif user_input == 'b' 
-  user_pos = user_pos.zip(b).map { |x, y| x + y }
-end #same thing if input is b...etc
+board[user_x][user_y] = character.chars[0].upcase
```

Comment:
Here's where I initially set the player's position on the board to their character's initial. This is a little confusing because the y comes first. That's because we have an array of arrays. So the first set of arrays are the rows, the second set is the array of columns in that row. That means that when you're dealing with the board, the first number will be the y/row, second will be the x/column in the row specified.